acceptance/Rakefile

```diff
@@ -120,143 +107,60 @@ EOS
-def list_preserved_configurations(secs_ago = ONE_DAY_IN_SECS)
-  preserved = {}
-  Dir.glob('log/*_*').each do |dir|
-    preserved_config_path = "#{dir}/preserved_config.yaml"
-    yesterday = Time.now - secs_ago.to_i
-    if preserved_config = File.exists?(preserved_config_path)
-      directory = File.new(dir)
-      if directory.ctime > yesterday
-        hosts = []
-        preserved_config = YAML.load_file(preserved_config_path).to_hash
-        preserved_config['HOSTS'].each do |hostname,values|
-          hosts << "#{hostname}: #{values['platform']}, #{values['roles']}"
-        end
-        preserved[hosts] = directory.to_path
-      end
-    end
-  end
-  preserved.map { |k,v| [v,k] }.sort { |a,b| a[0] <=> b[0] }.reverse
+  sh("beaker", *args)
 end
 
-def list_preserved_hosts(secs_ago = ONE_DAY_IN_SECS)
+def preserved_hosts(secs_ago = ONE_DAY_IN_SECS)
   hosts = Set.new
-  Dir.glob('log/**/pre*suite*run.log').each do |log|
-    yesterday = Time.now - secs_ago.to_i
-    File.open(log, 'r') do |file|
-      if file.ctime > yesterday
-        file.each_line do |line|
-          matchdata = /^(\w+)(?:\.[\w.]+)? \(.*?\) \d\d:\d\d:\d\d\$/.match(line.encode!('UTF-8', 'UTF-8', :invalid => :replace))
-          hosts.add(matchdata[1]) if matchdata
+  # if secs_ago = 0, just use latest log
+  do_only_log_latest = secs_ago == 0
+  dirs = do_only_log_latest ? Dir.glob('log/latest') : Dir.glob('log/*')
+  yesterday = do_only_log_latest ? Time.now - ONE_DAY_IN_SECS : Time.now - secs_ago.to_i
+  dirs.each do |dir|
+    if dir !~ /^log\/latest$/ || do_only_log_latest
+    hosts_dirs = do_only_log_latest ? Dir.glob(dir) : Dir.glob("#{dir}/*")
+      hosts_dirs.each do |date_dir|
+        preserved_config_path = "#{date_dir}/#{SUTS_FILENAME}"
+        if preserved_config = File.exists?(preserved_config_path)
+          File.open(preserved_config_path, 'r') do |file|
+            if file.ctime > yesterday
+              file.each_line do |line|
+                # match every line without [-]
+                matchdata = /^((?!\[-\]).)*$/.match(line)
+                hosts.add(line) if matchdata
+              end
+            end
+          end
         end
       end
     end
   end
   hosts
 end
 
-def release_hosts(hosts = nil, secs_ago = ONE_DAY_IN_SECS)
-  secs_ago ||= ONE_DAY_IN_SECS
-  hosts ||= list_preserved_hosts(secs_ago)
+def vmpooler_token
+  `grep --extended --only-matching 'vmpooler_token: .*' ${HOME}/.fog`.split(' ')[1]
+end
 
+def release_hosts(hosts = nil)
   hosts.each do |h|
-    hostname = h.split('.').first
+    hostname = h.match(/\w{15}\.delivery\.puppetlabs\.net/)[0]
     puts "Releaseing '#{hostname}'"
-    puts `curl -X DELETE --url http://vcloud.delivery.puppetlabs.net/vm/#{hostname}`
+    shh_curl = 'curl --silent --show-error'
+    token_command = "-H X-AUTH-TOKEN:#{vmpooler_token}"
+    vcloud_url = "http://vcloud.delivery.puppetlabs.net/vm/#{hostname}"
+    puts `#{shh_curl} -X DELETE --url #{vcloud_url} #{token_command}`
```

Comment:
I don't think this is new in this PR (so not blocking), but if I kill VMs that I previously allocated, then they are successfully destroyed, but subsequent `list_hosts` still includes them:

```
$ bundle exec rake ci:list_hosts
#0: 2017-02-22 23:42:01	[+]	vmpooler	el-7-x86_64	y64f0zs6p1a6j76.delivery.puppetlabs.net (redhat7-64-1)
#1: 2017-02-22 23:42:01	[+]	vmpooler	el-7-x86_64	j55eve0tn2oxo8b.delivery.puppetlabs.net (redhat7-64-2)
$ bundle exec rake ci:kill_hosts
Releaseing 'y64f0zs6p1a6j76.delivery.puppetlabs.net'
{
  "ok": true
}
Releaseing 'j55eve0tn2oxo8b.delivery.puppetlabs.net'
{
  "ok": true
}
$ bundle exec rake ci:list_hosts
#0: 2017-02-22 23:42:01	[+]	vmpooler	el-7-x86_64	y64f0zs6p1a6j76.delivery.puppetlabs.net (redhat7-64-1)
#1: 2017-02-22 23:42:01	[+]	vmpooler	el-7-x86_64	j55eve0tn2oxo8b.delivery.puppetlabs.net (redhat7-64-2)
```